### PR TITLE
Created SmoothImageDiv general compnent, used on homepage hero.

### PR DESCRIPTION
**Pull request by jcdesimp**
The SmoothImageDiv component will make sure a div background image is fully loaded before fully displaying it.

The SmoothImageDiv component takes an image url, className, and a delay in milliseconds.

The component will apply the "loading" class to the div until the background image has been loaded. The transition and "loading" properties will need to be set in your style sheet.

It also allows a delay to be specified to ensure different SmoothImageDivs will fade in at the same time.

The images on the homepage will set opacity to 0 until the images are loaded, with a delay of 400 ms.

Preview:

smoothimageloading

### CODE REVIEW

**Comment by 7imbrook** [comment]
@jcdesimp  What does a shorter fad in look like?

**Comment by vuzDylan** [comment]
agree a shorter fad might be better but I think it's kinda cool

**Comment by kristenmills** [comment]
Fancy.

File: app/js/components/general/SmoothImageDiv.jsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        updated: Date.now(),
+      }), this.props.delayMs);
+    }
+    const img = new Image();
+    img.onload = () => this.setState({ loading: false });
+    img.src = this.props.imageUrl;
+  }
+
+  render() {
+    const style = Object.assign({}, this.props.style, {
```
Comment (vuzDylan) [inline app/js/components/general/SmoothImageDiv.jsx]: Use a spread we don't use object.assign anywhere else

File: app/js/components/general/SmoothImageDiv.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    return (
+      <div
+        className={classes.join(' ')}
+        style={style}
+      />
+    );
+  }
+}
+
+SmoothImageDiv.propTypes = {
```
Comment (kristenmills) [inline app/js/components/general/SmoothImageDiv.jsx]: insid the class body,

`static proptypes = { ... }`

same for default props.

File: app/js/components/general/SmoothImageDiv.jsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    img.onload = () => this.setState({ loading: false });
+    img.src = this.props.imageUrl;
+  }
+
+  render() {
+    const style = Object.assign({}, this.props.style, {
+      backgroundImage: `url(${this.props.imageUrl})`,
+    });
+
+    const classes = [this.props.className];
```
Comment (kristenmills) [inline app/js/components/general/SmoothImageDiv.jsx]: Ues https://github.com/JedWatson/classnames
it's already a dependency.

File: app/js/components/home/Hero.jsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    super(props);
+    this.state = {
+      img1Loaded: false,
+      img2Loaded: false,
+    };
+
+    this.handleImageLoad = this.handleImageLoad.bind(this);
+  }
+
+  handleImageLoad(key) {
```
Comment (vuzDylan) [inline app/js/components/home/Hero.jsx]: handleImageLoad = (key) => {

File: app/js/components/home/Hero.jsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+class Hero extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      img1Loaded: false,
+      img2Loaded: false,
+    };
+
+    this.handleImageLoad = this.handleImageLoad.bind(this);
```
Comment (vuzDylan) [inline app/js/components/home/Hero.jsx]: delete

**Pull request merged**